regressors/include/iDynTree/Regressors/DynamicsRegressorGenerator.h

```diff
@@ -0,0 +1,312 @@
+ *
+ */
+class DynamicsRegressorGenerator {
+private:
+    struct DynamicsRegressorGeneratorPimpl;
+    DynamicsRegressorGeneratorPimpl * pimpl;
+
+public:
+
+    /**
+     * @name Constructor/Destructor
+     */
+    //@{
+
+    /**
+     * Constructor
+     *
+     */
+    DynamicsRegressorGenerator();
+
+    /**
+     * Destructor
+     *
+     */
+    virtual ~DynamicsRegressorGenerator();
+    //@}
+
+    /**
+     * @name Model loading and definition methods
+     * This methods are used to define the structure of your regressor, i.e. load the model
+     * and define the desired regressors.
+     */
+    //@{
+
+    /**
+     * Load the model of the robot and the sensors from an external file.
+     *
+     * @param filename path to the file to load
+     * @param filetype type of the file to load, currently supporting only urdf type.
+     *
+     */
+    bool loadRobotAndSensorsModelFromFile(const std::string & filename, const std::string & filetype="detect");
+
+    /**
+     * Load the model of the robot and the sensors from a string.
+     *
+     * @param filename string containg the model of the robot.
+     * @param filetype type of the file to load, currently supporting only urdf type.
+     *
+     */
+    bool loadRobotAndSensorsModelFromString(const std::string & modelString, const std::string & filetype="detect");
+
+    // Coming soon: load model and sensor from iDynTree data structures
+    //loadRobotAndSensorsModel()
+
+    /**
+     * Load the regressor structure from a configuration file.
+     *
+     * Currently only the subtreeBaseDynamics regressor type is supported.
+     * For load the regressor of the subtree that has the `r_arm` as the FTSensorLink,
+     * you can pass the following xml string:
+     * ~~~
+     * <regressor>
+     *     <subtreeBaseDynamics>
+     *         <FTSensorLink>r_arm</FTSensorLink>
+     *     </subtreeBaseDynamics>
+     * </regressor>
+     * ~~~
+     *
+     * For loading the regressor composed of the two subtrees, one that has the `l_foot` as FTSensorLink,
+     * and the other that has `l_ankle_2` and `l_hip_3` has FTSensorLink, you can pass the following xml string:
+     *
+     * ~~~
+     * <regressor>
+     *     <subtreeBaseDynamics>
+     *         <FTSensorLink>l_foot</FTSensorLink>
+     *     </subtreeBaseDynamics>
+     *     <subtreeBaseDynamics>
+     *         <FTSensorLink>l_ankle_2</FTSensorLink>
+     *         <FTSensorLink>l_hip_3</FTSensorLink>
+     *     </subtreeBaseDynamics>
+     * </regressor>
+     * ~~~
+     *
+     * For subtreeBaseDynamics subregressor, you specify the frame of expressions (i.e
+     * with respect to which point and to which orientation the regressor equation will be
+     * written) using the `<frame>` or `<sensorFrame>` tag.
+     *
+     * @param filename path to the file to load
+     * @param filetype type of the file to load, currently supporting only urdf type.
```

Comment:
For now I actually get rid of the filetype for the regressor structure.. in the future we could support also other format (json? ini?) but for now I simply assume that is the xml format documented in the comments.